src/SendGrid/SendGridClient.cs

```diff
@@ -73,6 +74,15 @@ public SendGridClient(SendGridClientOptions options)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendGridClient"/> class.
+        /// </summary>
+        /// <param name="options">An <see cref="IOptions{SendGridClientOptions}"/> instance specifying the configuration to be used with the client.</param>
+        public SendGridClient(IOptions<SendGridClientOptions> options)
+            : this(options.Value)

```

Comment:
`options?.Value ?? throw new ArgumentNullException(nameof(options))` here as well?

Reply:
Wouldn't it make more sense to check for null in `SendGridClient(HttpClient httpClient, SendGridClientOptions options)` in that case, and just pass `options?.Value` in the `this` calls?

Reply:
Agreed. I've committed the changes above and I'm positive they meet your requirements. Let me know what you think.